### ISSUE TITLE
Redundant instructions on AVX2 when using static mask with half lanes disabled

### ISSUE THREAD

**Issue by dbabokin** [ispc/ispc#1499]
The issue was discovered by @pbrubaker. Here's the source code:

```
struct FVector4 {
    float V[4];
};

unmasked uniform FVector4 Mul1(const uniform FVector4 &A, const uniform FVector4 &B) {
    uniform FVector4 Result;
    foreach(i = 0 ... 4) {
        Result.V[i] = A.V[i] * B.V[i];
    }
    return Result;
}
```

With ISPC 1.11.0 it produces the following assembler for AVX2:
```
	vmovaps	LCPI0_0(%rip), %ymm0    ## ymm0 = [4294967295,4294967295,4294967295,4294967295,0,0,0,0]
	vmaskmovps	(%rdi), %ymm0, %ymm1
	vmaskmovps	(%rsi), %ymm0, %ymm2
	vmovaps	%xmm1, %xmm1
	vmovaps	%xmm2, %xmm2
	vmulps	%ymm2, %ymm1, %ymm1
	vmaskmovps	%ymm1, %ymm0, -16(%rsp)
```

Note that ``vmovaps`` instructions are redundant.

**Comment by dbabokin** [comment]
Here's a fix for this issue in LLVM trunk (9.0):
http://llvm.org/viewvc/llvm-project?view=revision&revision=362675

We should try applying it to LLVM 8.0

**Comment by dbabokin** [comment]
Fixed.
